extensions.rst

```diff
@@ -0,0 +1,13 @@
+.. _extensions:
+
+Write a CPython extension
+=========================
+
+In this section, we could explain how to write a CPython extension with the C language, but the topic can take a complete book.
```

Comment:
I think the goal of this section should be to equip folks with the tools they need to work on the existing extension modules in the reference implementation, and to add new ones if necessary.

Essential pointers:

- link to the full C API docs: https://docs.python.org/dev/c-api/
- link to PEP 399 specifically in relation to accelerator modules and testing both implementations